diff -ruaN a/drivers/kernelsu/policy/app_profile.c b/drivers/kernelsu/policy/app_profile.c
--- a/drivers/kernelsu/policy/app_profile.c	2026-04-10 23:32:37.592957800 +0800
+++ b/drivers/kernelsu/policy/app_profile.c	2026-04-10 23:36:59.998041200 +0800
@@ -63,7 +63,12 @@
     put_group_info(group_info);
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 12, 0)
+void put_seccomp_filter(struct task_struct *tsk);
+#define seccomp_filter_release put_seccomp_filter
+#else
 void seccomp_filter_release(struct task_struct *tsk);
+#endif
 
 static void disable_seccomp(void)
 {
@@ -372,4 +377,4 @@
     pr_info("cmd_su: privilege escalation completed for UID: %d, PID: %d\n",
             target_uid, target_pid);
 }
-#endif
+#endif
\ No newline at end of file
