--- a/drivers/kernelsu/kpm/kpm.c	2026-06-06 01:49:04.000000000 +0000
+++ b/drivers/kernelsu/kpm/kpm.c	2026-06-06 02:36:51.993619473 +0000
@@ -132,14 +132,14 @@
             goto exit;
         }
 
-        if (!access_ok(arg1, 255)) {
+        if (!access_ok(VERIFY_READ, (void __user *)arg1, 255)) {
             goto invalid_arg;
         }
 
         strncpy_from_user((char *)&kernel_load_path, (const char *)arg1, 255);
 
         if (arg2 != 0) {
-            if (!access_ok(arg2, 255)) {
+            if (!access_ok(VERIFY_READ, (void __user *)arg2, 255)) {
                 goto invalid_arg;
             }
 
@@ -158,7 +158,7 @@
             goto exit;
         }
 
-        if (!access_ok(arg1, sizeof(kernel_name_buffer))) {
+        if (!access_ok(VERIFY_READ, (void __user *)arg1, sizeof(kernel_name_buffer))) {
             goto invalid_arg;
         }
 
@@ -178,7 +178,7 @@
             goto exit;
         }
 
-        if (!access_ok(arg1, sizeof(kernel_name_buffer))) {
+        if (!access_ok(VERIFY_READ, (void __user *)arg1, sizeof(kernel_name_buffer))) {
             goto invalid_arg;
         }
 
@@ -189,7 +189,7 @@
         sukisu_kpm_info((const char *)&kernel_name_buffer, (char *)&buf,
                         sizeof(buf), &size);
 
-        if (!access_ok(arg2, size)) {
+        if (!access_ok(VERIFY_WRITE, (void __user *)arg2, size)) {
             goto invalid_arg;
         }
 
@@ -204,7 +204,7 @@
             goto exit;
         }
 
-        if (!access_ok(arg2, len)) {
+        if (!access_ok(VERIFY_WRITE, (void __user *)arg1, len)) {
             goto invalid_arg;
         }
 
@@ -222,11 +222,11 @@
         char kpm_name[KPM_NAME_LEN] = { 0 };
         char kpm_args[KPM_ARGS_LEN] = { 0 };
 
-        if (!access_ok(arg1, sizeof(kpm_name))) {
+        if (!access_ok(VERIFY_READ, (void __user *)arg1, sizeof(kpm_name))) {
             goto invalid_arg;
         }
 
-        if (!access_ok(arg2, sizeof(kpm_args))) {
+        if (!access_ok(VERIFY_READ, (void __user *)arg2, sizeof(kpm_args))) {
             goto invalid_arg;
         }
 
@@ -286,13 +286,13 @@
         return -EFAULT;
     }
 
-    if (!access_ok(cmd.control_code, sizeof(int))) {
+    if (!access_ok(VERIFY_READ, (void __user *)cmd.control_code, sizeof(int))) {
         pr_err("kpm: invalid control_code pointer %px\n",
                (void *)cmd.control_code);
         return -EFAULT;
     }
 
-    if (!access_ok(cmd.result_code, sizeof(int))) {
+    if (!access_ok(VERIFY_WRITE, (void __user *)cmd.result_code, sizeof(int))) {
         pr_err("kpm: invalid result_code pointer %px\n",
                (void *)cmd.result_code);
         return -EFAULT;
